### PR TITLE
Fix missing license headers and add auto-check

### PR DESCRIPTION
**Pull request by daviddrysdale**
### Checklist

- [ ] Pull request affects core Oak functionality (e.g. runtime, SDK, ABI)
  - [ ] I have written tests that cover the code changes.
  - [ ] I have checked that these tests are run by [Cloudbuild](cloudbuild.yaml)
  - [ ] I have updated [documentation](docs/) accordingly.
  - [ ] I have raised an [issue](https://github.com/project-oak/oak/issues) to
        cover any TODOs and/or unfinished work.
- [ ] Pull request includes prototype/experimental work that is under
      construction.

### CODE REVIEW

File: examples/abitest/module_0/rust/src/proto/mod.rs [hunk 1]
```diff
@@ -1,2 +1,5 @@
+//! Auto-generated code for processing protocol buffer message definitions and
+//! gRPC service definitions.
```
Comment (tiziano88) [inline examples/abitest/module_0/rust/src/proto/mod.rs]: Is this code __actually__ auto-generated? The other day @ipetr0v and I spent a long time trying to figure it out, and we arrived to the conclusion that it is not, or that we don't understand what is generating it. @daviddrysdale could you help us solve the mystery?

File: scripts/temp.sh [hunk 2]
```diff
@@ -0,0 +1,19 @@
+#!/usr/bin/env bash
```
Comment (tiziano88) [inline scripts/temp.sh]: Is this file leftover?

File: scripts/check_license [hunk 3]
```diff
@@ -0,0 +1,37 @@
+#!/bin/bash
+#
+# Checks that source files have the Apache License header.
+# Automatically skips generated files.
+set -eu
```
Comment (tiziano88) [inline scripts/check_license]: Could you source `scripts/common` here like we do in other scripts?

File: scripts/temp.sh (hunk 2 above)
Comment (daviddrysdale) [reply scripts/temp.sh]: Oops, dinged.

File: scripts/check_license (hunk 3 above)
Comment (daviddrysdale) [reply scripts/check_license]: Done.

File: examples/abitest/module_0/rust/src/proto/mod.rs (hunk 1 above)
Comment (daviddrysdale) [reply examples/abitest/module_0/rust/src/proto/mod.rs]: Hmm, I think you're right.  I guess the comment means that the sub-modules listed below are auto-generated, not that this index of them is.

I've moved the various `mod.rs` files to have a license header.

Comment (tiziano88) [reply examples/abitest/module_0/rust/src/proto/mod.rs]: :100: this is valuable IP

**Pull request merged**